wasmtime-jni/src/io_github_kawamuray_wasmtime_Config/imp.rs

```diff
@@ -0,0 +1,198 @@
+use crate::errors;
+use crate::interop;
+use crate::io_github_kawamuray_wasmtime_Config::JniConfig;
+use jni::objects::{JClass, JObject, JString, JValue};
+use jni::sys::{jboolean, jlong, jobject};
+use jni::{self, JNIEnv};
+use std::path::Path;
+use std::result::Result;
+use wasmtime::{Config, OptLevel, ProfilingStrategy, Strategy};
+
+pub(super) struct JniConfigImpl;
+
+impl<'a> JniConfig<'a> for JniConfigImpl {
+    type Error = errors::Error;
+    fn cache_config_load(
+        env: &JNIEnv,
+        this: JObject,
+        path: JObject,
+    ) -> Result<jobject, Self::Error> {
+        let mut config = interop::get_inner::<Config>(env, this)?;
+        let re = env.call_method(path, "toString", "(V)Ljava/lang/String;", &[]);
+        let j_obj = re.unwrap().l().unwrap();
+        let j_str = JString::from(j_obj);
+        let path_java_str = env.get_string(j_str).unwrap();
```

Comment:
@kawamuray I met a problem while using `?`:
```
error[E0277]: `?` couldn't convert the error to `errors::Error`
  --> src\io_github_kawamuray_wasmtime_Config\imp.rs:26:54
   |
26 |         let path_str = env.get_string(path)?.to_str()?;
   |                                                      ^ the trait `From<Utf8Error>` is not implemented for `errors::Error`
   |
   = note: the question mark operation (`?`) implicitly performs a conversion on the error value using the `From` trait
   = help: the following implementations were found:
             <errors::Error as From<PoisonError<G>>>
             <errors::Error as From<WasiCtxBuilderError>>
             <errors::Error as From<anyhow::Error>>
             <errors::Error as From<jni::errors::Error>>
             <errors::Error as From<std::io::Error>>
   = note: required by `from`
```
So I use this instead:
```rust
fn cache_config_load_native(
    env: &JNIEnv,
    this: JObject,
    path: JString,
) -> Result<jobject, Self::Error> {
    let mut config = interop::get_inner::<Config>(env, this)?;
    let path_j_str = env.get_string(path)?;
    //the trait `From<Utf8Error>` is not implemented for `errors::Error`
    //So the `?` operator cannot be used at here
    let path_str = path_j_str.to_str().unwrap();
    config.cache_config_load(Path::new(path_str))?;
    return Ok(this.into_inner());
}
```
Is it OK?

Reply:
Yes, `Utf8Error` is not the subject of auto conversion to the expected return type, and I think it happens in extremely rare situation. So okay to unwrap, but as I commented we should prefer to use `.expect()` instead of `.unwrap()` to provide better diagnostic when it actually happens.